R/simulate_linreg.R

```diff
@@ -277,60 +213,41 @@ sim_sumstats <- function(RL, ngwas, beta, h2ge) {
 # The following is modified using chatgpt4 from
 # https://github.com/xueweic/fine-mcoloc/blob/b7aa9d27b0be13044224a4b2207717e74611d449/dsc/phenotype_simulation.r
 
-#' Simulate Multiple Traits from Genotype and Effect Sizes
+#' Simulate Multiple Traits from Genotype
 #'
-#' This function simulates multiple traits (phenotypes) based on genotype data, 
-#' a matrix of effect sizes, and heritability. It allows specifying if the heritability
+#' This function simulates multiple traits (phenotypes) based on genotype data, the rows of output will represent each trait, and the columns are each subjects' phenotype (trait)
+#' and heritability. It allows specifying if the heritability
 #' is total or per-SNP, optionally scales the phenotypes, and can handle residual correlations.
 #' per eQTL heritability is 0.05 to 0.07 according to https://www.nature.com/articles/ng.3506
 #'
 #' @param G Genotype matrix.
 #' @param B Matrix of effect sizes for multiple traits.
 #' @param h2g Heritability (proportion of variance explained by genetics).
 #' @param is_h2g_total Logical indicating if h2g is total (TRUE) or per-SNP (FALSE).
-#' @param max_h2g Maximum heritability allowed in the model (default is 0.8).
 #' @param residual_corr Matrix of residual correlations (NULL for independent samples).
 #' @param scale_Y Logical indicating if the phenotypes should be scaled (default is FALSE).
 #' @return A list containing the simulated phenotypes (`Y`) and residual variance (`residual_var`).
 #' @importFrom MASS mvrnorm
 #' @export
 #'
 #' @examples
-#' G <- matrix(rnorm(1000), nrow = 100, ncol = 10) # Random genotype matrix
```

Comment:
You seem to have made too many changes (especially deletion of features) here without justification. It is okay to remove scaling for Y. But this is a multivariate setting. You are not supposed to loop through columns of Y. Rather you need to think of this as a matrix. This is not just for your project. You make it generic also for other projects where we allow for residual correlation. Your project should be a specifical case in this. 

of course you can use your special case to test and verify things are correct. 

Please revert your changes --- try to stick to the original logic here. It's okay to remove scale_Y parameter. But let's do a proper multivariate matrix based simulation.